bluepyopt/ephys/mechanisms.py

```diff
@@ -85,6 +85,9 @@ def __init__(
         self.preloaded = preloaded
         self.cell_model = None
         self.deterministic = deterministic
+        # If deterministic is False, it might get changed to True.
```

Comment:
I think we need a bit better description here. It's a bit weird code, so I think the use case should be documented well

Reply:
'One might want to temporarily set a stochastic mechanism to deterministic to run a few protocols, and then turn it back to stochastic mode to run some other protocols. The always_deterministic attribute remembers the mechanisms that were set to be deterministic or stochastic when created. If one wants to turn the stochasticity back on, after temporarily forcing the mechanisms to be deterministic, one can set the deterministic attribute to the value of always_deterministic using the self.set_stochasticity(True) method'

Would a comment like this be acceptable?